src/main/java/com/pusher/client/channel/impl/BaseChannel.java

```diff
@@ -111,32 +110,24 @@ public String toUnsubscribeMessage() {
         return GSON.toJson(new UnsubscribeMessage(getName()));
     }
 
-    @Override
-    public PusherEvent prepareEvent(String event, String message) {
-        return GSON.fromJson(message, PusherEvent.class);
+    public void emit(PusherEvent pusherEvent) {
+        final Set<SubscriptionEventListener> listeners = getInterestedListeners(pusherEvent.getEventName());
+        if (listeners != null) {
+            for (final SubscriptionEventListener listener : listeners) {
+                factory.queueOnEventThread(() -> listener.onEvent(pusherEvent));
+            }
+        }
     }
 
     @Override
-    public void onMessage(String event, String message) {
-        if (event.equals(SUBSCRIPTION_SUCCESS_EVENT)) {
+    public void handleEvent(PusherEvent event) {
+        if (event.getEventName().equals(SUBSCRIPTION_SUCCESS_EVENT)) {
             updateState(ChannelState.SUBSCRIBED);
-        }else if (event.equals(SUBSCRIPTION_COUNT_EVENT)) {
-            handleSubscriptionCountEvent(message);
         } else {
-            final Set<SubscriptionEventListener> listeners = getInterestedListeners(event);
-            if (listeners != null) {
-                final PusherEvent pusherEvent = prepareEvent(event, message);
-                if (pusherEvent != null) {
-                    for (final SubscriptionEventListener listener : listeners) {
-                        factory.queueOnEventThread(new Runnable() {
-                                @Override
-                                public void run() {
-                                    listener.onEvent(pusherEvent);
-                                }
-                            });
-                    }
-                }
+            if (event.getEventName().equals(SUBSCRIPTION_COUNT_EVENT)) {
+                handleSubscriptionCountEvent(event);
             }
+            emit(event);
```

Comment:
Processing the internal subscription count event further is a change in behaviour that is not required to fix the reported issue if I understand correctly. Do we need this?

Reply:
It's used to update the internal subscriptionCount value, which provides for the getCount() method

Reply:
That is done in the `handleSubscriptionCountEvent`.

What I'm pointing out is
* Before this PR, `handleEvent` stopped after calling `handleSubscriptionCountEvent`. The only event processed further would be the public subscription count event.
* With this PR, we call `emit` on the internal event after calling `handleSubscriptionCountEvent`. The equivalent behaviour wasn't there before. I'm not sure it is needed

Reply:
Ok changed it